### PR TITLE
Support for OpenTracing 0.33.0

### PR DESCRIPTION
**Pull request by agarbutt**
Support for #18

### CODE REVIEW

**Comment by jeffalder** [comment]
Hi @agarbutt , Thanks for the contribution! I'm going to take a look at this.

File: src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java [hunk 1]
```diff
@@ -42,16 +43,18 @@ default Output handleRequest(Input input, Context context) {
...
-        ResponseParser.parseResponse(output, scope.span());
+        ResponseParser.parseResponse(output, span);
         return output;
       } catch (Throwable throwable) {
-        scope.span().log(SpanUtil.createErrorAttributes(throwable));
+        span.log(SpanUtil.createErrorAttributes(throwable));
         throw throwable;
       }
+    } finally {
+      span.finish();
```
Comment (jeffalder) [inline src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java:57]: am I the only one annoyed that `Span` doesn't implement `Closeable`, so you could _really_ use try-with-resources?

**Comment by jasonjkeller** [comment]
We should also update the AWS Lambda Tracer to support OpenTracing 0.33 in conjunction with the this PR for the SDK. https://newrelic.atlassian.net/browse/JAVA-5898

File: src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package com.newrelic.opentracing.aws;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+
+public class EnhancedSpanBuilder {
```
[the remark this answers is not in the document]
Comment (agarbutt) [reply src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java:12]: Thanks. I had / have some other ideas, but if you're moving to OpenTelemetry, i'd rather get this minimally supported and then invest there. 👍

File: src/main/java/com/newrelic/opentracing/aws/SpanUtil.java [hunk 3]
```diff
@@ -6,29 +6,29 @@
...
+  static <Input> Span buildSpan(
+      Input input, Context context, Tracer tracer, SpanContext spanContext) {
+    return EnhancedSpanBuilder.basedOn(tracer, "handleRequest")
+        .asChildOf(spanContext)
+        .withTag("aws.requestId", context.getAwsRequestId())
+        .withTag("aws.lambda.arn", context.getInvokedFunctionArn())
+        .optionallyWithTag(
+            "aws.lambda.eventSource.arn", EventSourceParser.parseEventSourceArn(input))
+        .optionallyWithTag(
+            "aws.lambda.coldStart", TracingRequestHandler.isColdStart.getAndSet(false))
```
[the remark this answers is not in the document]
Comment (agarbutt) [reply src/main/java/com/newrelic/opentracing/aws/SpanUtil.java]: Great catch! This is a bad refactoring. See my updated commit.

File: src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package com.newrelic.opentracing.aws;
```
[the remark this answers is not in the document]
Comment (agarbutt) [reply src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java:6]: Done.

File: src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java (hunk 1 above)
Comment (agarbutt) [reply src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java:57]: yeah, the choice made to move away from making the Span Auto-Closable was a surprise to me.

**Comment by agarbutt** [comment]
Thanks for the feedback @jeffalder. I believe I have addressed the concerns you raised. 

I'd be happy to help contribute, if that would be helpful, @jasonjkeller

File: src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffalder) [reply src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java:12]: Agreed

**Comment by jeffalder** [comment]
We'll have to ship an update to the other side of this to match up OT versions.

**Pull request merged**